addons/web/static/src/views/form/form_controller.js

```diff
@@ -263,6 +251,20 @@ export class FormController extends Component {
         return this.model.root.data.display_name || this.env._t("New");
     }
 
+    async onPagerUpdate({ offset, resIds }) {
+        await this.model.root.askChanges(); // ensures that isDirty is correct
+        let canProceed = true;
+        if (this.model.root.isDirty) {
+            canProceed = await this.model.root.save({
+                stayInEdition: true,
+                useSaveErrorDialog: true,
+            });
+        }
+        if (canProceed) {
+            return this.model.load({ resId: resIds[offset] });
+        }
+    }
+
```

Comment:
I had to extract this to have a hook in the test when "update" happens (e.g. when switching to the next record). Maybe I can also do it by patching "model.load". If you prefer me patching "model.load", where can I find the class for this "model" instance?

Reply:
yep, I umderstand. It's not a big deal either way. But extracting the onPagerUpdate code sounds fine in any case, since it is not trivial, it is not a bad idea not to inline it